python/tank_vendor/shotgun_base/utils.py

```diff
@@ -18,6 +18,34 @@
 from .log import get_shotgun_base_logger
 log = get_shotgun_base_logger()
 
+
+def with_cleared_umask(func):
+    """
+    Decorator which clears the umask for a method.
+
+    The umask controls default permissions for files
+    and folders. By setting the umask to zero, we can control
+    permissions.
+
+    Example use:
+
+    @with_cleared_umask
+    def my_io_method(param1, param2):
+
```

Comment:
I wonder if a context manager wouldn't be better? it seems this is the kind of operation what you want to be as tight as possible. Using it with the 'with' statement would probably result in a shorter window for the umask change. What do you think?